trunk/src/kernel/srs_kernel_mp4.cpp

```diff
@@ -5431,15 +5430,14 @@ srs_error_t SrsMp4Decoder::initialize(ISrsReadSeeker* rs)
     off_t offset = -1;
     
     while (true) {
-        SrsMp4Box* box = NULL;
-        SrsAutoFree(SrsMp4Box, box);
-        
-        if ((err = load_next_box(&box, 0)) != srs_success) {
+        SrsMp4Box* box_raw = NULL;
+        if ((err = load_next_box(&box_raw, 0)) != srs_success) {
```

Comment:
This is bug here, `box_raw` is managed by `SrsUniquePtr` here and `SrsAutoFree` inside `load_next_box`, line L5655. so this pointer will be released twice. It's ok for both of them are managed by `SrsAutoFree`, but not this way.